### PR TITLE
SPCTR-3203 Fix: Forms - Resolved an issue of wherein forms block not working inside astra custom layout, when we use it inside container.

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

- https://brainstormforce.atlassian.net/browse/SPCTR-3203

### Screenshots

- https://d.pr/v/cEFAZi

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

- Added to the card

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: blocks-config/forms/class-uagb-forms.php [hunk 1]
```diff
@@ -260,9 +261,11 @@ public function process_forms() {
 						$post_content = $custom_post->post_content;
 						if ( has_block( 'uagb/forms', $post_content ) ) {
 							$blocks = parse_blocks( $post_content );
-							if ( ! empty( $blocks[0]['attrs']['block_id'] ) ) {
-								$block_id                 = $blocks[0]['attrs']['block_id'];
+							if ( ! empty( $blocks ) && is_array( $blocks ) ) {
 								$current_block_attributes = $this->recursive_inner_forms( $blocks, $block_id );
+								if ( is_array( $current_block_attributes ) && $current_block_attributes['block_id'] === $block_id ) {
```
Comment (adi3890) [inline blocks-config/forms/class-uagb-forms.php:266]: any possibility of $current_block_attributes be empty

File: blocks-config/forms/class-uagb-forms.php [hunk 2]
```diff
@@ -186,7 +186,7 @@ public function process_forms() {
 				wp_send_json_error( 400 );
 			}
 			$current_block_attributes = false;
-			$block_id                 = sanitize_text_field( $_POST['block_id'] );
+			$block_id                 = (int) sanitize_text_field( $_POST['block_id'] );
```
Comment (adi3890) [inline blocks-config/forms/class-uagb-forms.php:189]: please check if we can use absint in place of sanitizing and converting value

**Review by adi3890 — APPROVED** [review]

File: blocks-config/forms/class-uagb-forms.php (hunk 1 above)
Comment (AbhijitNage123) [reply blocks-config/forms/class-uagb-forms.php:266]: No @adi3890

File: blocks-config/forms/class-uagb-forms.php (hunk 2 above)
Comment (AbhijitNage123) [reply blocks-config/forms/class-uagb-forms.php:189]: Done

**Review by adi3890 — APPROVED** [review]

**Review by rajmohanreddy774 — APPROVED** [review]
Looks good to me

**Pull request merged**